docs/.vitepress/blog.data.ts

```diff
@@ -9,7 +9,10 @@ interface Post {
   }
 }
 
-declare const data: Post[]
+// TODO: the exact typed data loader pattern is not supported by rolldown/oxc transform
+// https://vitepress.dev/guide/data-loading#typed-data-loaders
+// eslint-disable-next-line import/no-mutable-exports
+let data!: Post[]
 export { data }
```

Comment:
Vitepress's typed data loader pattern looks like a bit broken concept. Esbuild-based config loader somehow eliminated to `export {}`, but rolldown/oxc transpiles it to `export { data }` and that causes a runtime error.

- esbuild https://esbuild.github.io/try/#YgAwLjI1LjEAeyBmb3JtYXQ6ICdlc20nIH0AZQBlbnRyeS50cwBkZWNsYXJlIGNvbnN0IGRhdGE6IFBvc3RbXQpleHBvcnQgeyBkYXRhIH0
- rolldown https://stackblitz.com/edit/github-ga7ubbtm?file=src%2Ftest.ts,rolldown.config.mjs

Reply:
It seems it's related to https://github.com/rolldown/rolldown/issues/3048.

Reply:
Right, that one! It looks so familiar but I didn't remember where I saw. Indeed, this workaround looks cooler :smile: 

```ts
export declare const data: Post[]
```